docs/jax.experimental.pjit.rst

```diff
@@ -0,0 +1,9 @@
+jax.experimental.pjit module
```

Comment:
It'd be great to have the mesh ctxt manager and maybe even the class documented as well, if it's not already. Doesn't have to be this change though.

Reply:
It is documented in the `maps` module. We could lift it to `experimental` since it's common.